xc/device.ml

```diff
@@ -1986,6 +1986,11 @@ module Backend = struct
               )
 
           let qmp_event_thread () =
+            (* Add the existing qmp sockets first *)
+            List.iter (
+              fun x -> try Scanf.sscanf x "qmp-event-%d" (fun x -> add x) with _ -> ()
+            ) (Array.to_list (Sys.readdir "/var/run/xen"));
```

Comment:
Is this location kept in some constant where it should be fetched from rather than hardcoded here?

Reply:
There doesn't have this constant currently. 
It's also used by
xc/xenops_xc_main.ml
```
let make_var_run_xen () =
        Stdext.Unixext.mkdir_rec "/var/run/xen" 0o0755
```
Should I create one?

Reply:
Yes. If we use this location throughout the code base, there should be a constant for it.

Reply:
Thanks, submit a new commit https://github.com/xapi-project/xenopsd/pull/395/commits/7ee4cbc79e577a8588f62b6db161fc8f03a70d49

Reply:
Please dont't merge, Travis failed. I'm looking into it.